notion_client/helpers.py

```diff
@@ -124,3 +134,169 @@ def is_equation_rich_text_item_response(rich_text: Dict[Any, Any]) -> bool:
 def is_mention_rich_text_item_response(rich_text: Dict[Any, Any]) -> bool:
     """Return `True` if `rich_text` is a mention."""
     return rich_text.get("type") == "mention"
+
+
+def _format_uuid(compact_uuid: str) -> str:
+    """Format a compact UUID (32 chars) into standard format with hyphens."""
+    if len(compact_uuid) != 32:
+        raise ValueError("UUID must be exactly 32 characters")
+
+    return (
+        f"{compact_uuid[:8]}-{compact_uuid[8:12]}-{compact_uuid[12:16]}-"
+        f"{compact_uuid[16:20]}-{compact_uuid[20:]}"
+    )
+
+
+def extract_notion_id(url_or_id: str) -> Optional[str]:
+    """Extract a Notion ID from a Notion URL or return the input if it's already a valid ID.
+
+    Prioritizes path IDs over query parameters to avoid extracting view IDs instead of database IDs.
+
+    Args:
+        url_or_id: A Notion URL or ID string
+
+    Returns:
+        The extracted UUID in standard format (with hyphens) or None if invalid
+
+    Example:
+        >>> # Database URL with view ID - extracts database ID, not view ID
+        >>> extract_notion_id('https://notion.so/workspace/DB-abc123def456789012345678901234ab?v=viewid123')
+        'abc123de-f456-7890-1234-5678901234ab'  # database ID
+
+        >>> # Already formatted UUID
+        >>> extract_notion_id('12345678-1234-1234-1234-123456789abc')
+        '12345678-1234-1234-1234-123456789abc'
```

Comment:
We use markdown for our docstrings, not Google-style docstrings. Can we remove the Args block, say "Returns" as a real sentence and add the example in a code block?

Let's do something similar for the other docstrings as well.